src/ott/solvers/linear/sinkhorn.py

```diff
@@ -42,6 +44,21 @@
 
 __all__ = ["Sinkhorn", "SinkhornOutput", "solve"]
 
+ProgressCallbackFn = Callable[
+    [Tuple[np.ndarray, np.ndarray, np.ndarray, NamedTuple]], None]
```

Comment:
I think we can be a bit more specific in the 4th argument and require `SinkhornOutput`, in #284 the outputs will be unified nonetheless so we can defer after that PR is merged.